roboteq_diff_driver/src/driver.cpp

```diff
@@ -218,329 +24,132 @@ MainNode::MainNode() :
-  controller.write("^KD 2 0\r");
-
-  // set encoder mode (18 for feedback on motor1, 34 for feedback on motor2)
-  controller.write("^EMOD 1 18\r");
-  controller.write("^EMOD 2 34\r");
-
-  */
-
-  // enable watchdog timer (1000 ms)
-  controller.write("^RWD 250\r");
+  //watchdog 1000ms  
+  controller.write("^RWD 1000\r");
 
   // Limit acceleration
-  if (1) {
-    controller.write("^MAC 1 4000\r");
-    controller.write("^MAC 2 4000\r");
-    controller.write("^MDEC 1 4000\r");
-    controller.write("^MDEC 2 4000\r");
-    controller.write("^KP 1 100\r");
-    controller.write("^KP 2 100\r");
-    controller.write("^KD 1 0\r");
-    controller.write("^KD 2 0\r");
-    controller.write("^KI 1 20\r");
-    controller.write("^KI 2 20\r");
-  }
-  else {
-    controller.write("^MAC 1 20000\r");
-    controller.write("^MAC 2 205000\r");
-    controller.write("^MDEC 1 20000\r");
-    controller.write("^MDEC 2 20000\r");
-    controller.write("^KP 1 10\r");
-    controller.write("^KP 2 10\r");
-    controller.write("^KD 1 0\r");
-    controller.write("^KD 2 0\r");
-    controller.write("^KI 1 80\r");
-    controller.write("^KI 2 80\r");
+  controller.write("^MMOD 1 " + std::to_string(drive_mode) + "\r"); //set mode, 0: open loop, 5: torque
+  controller.write("^MMOD 2 " + std::to_string(drive_mode) + "\r");
+  controller.write("^BPR 1 " + std::to_string(override_torque) + "\r"); //override ramping
+  controller.write("^BPR 2 " + std::to_string(override_torque) + "\r");  
+  controller.write("^MXRPM 1 " + std::to_string(max_rpm) + "\r"); //max rpm
+  controller.write("^MXRPM 2 " + std::to_string(max_rpm) + "\r");
+
+
+  if(drive_mode == 5){
+    controller.write("^CLERD 1 0\r"); //disable closed loop error detection
+    controller.write("^CLERD 2 0\r");
+    controller.write("^BLSTD 1 0\r"); //disable stall detection
+    controller.write("^BLSTD 2 0\r");
+    controller.write("^ALIM 1 "+std::to_string(max_amps)+"\r"); //amps limit
+    controller.write("^ALIM 2 "+std::to_string(max_amps)+"\r");
+    controller.write("^MAC 1 50000\r"); //maximum acceleration
+    controller.write("^MAC 2 50000\r");
+    controller.write("^MDEC 1 50000\r"); //maximum deceleration
+    controller.write("^MDEC 2 50000\r");
   }
-  
   // set encoder counts (ppr)
   std::stringstream right_enccmd;
   std::stringstream left_enccmd;
   right_enccmd << "^EPPR 1 " << encoder_ppr << "\r";
   left_enccmd << "^EPPR 2 " << encoder_ppr << "\r";
   controller.write(right_enccmd.str());
   controller.write(left_enccmd.str());
-
   controller.flush();
 
-  ROS_INFO_STREAM("Subscribing to topic " << cmdvel_topic);
-  cmdvel_sub = nhLocal.subscribe(cmdvel_topic, 1000, &MainNode::cmdvel_callback, this);
-
+  ROS_INFO_STREAM("Subscribing to topic " << rpm_topic);
+  rpm_sub = nhLocal.subscribe(rpm_topic, 1000, &MainNode::rpm_callback, this);
 }
 
-void MainNode::cmdvel_loop()
+void MainNode::rpm_loop()
 {
 }
 
-void MainNode::cmdvel_run()
+void MainNode::rpm_run()
```

Comment:
same?

Reply:
This and previous both removed, note that they weren't being used beforehand either, I just renamed them to be in line with rest of code